### ISSUE TITLE
bash scripts for generalization experiments using task_arithmetic & ties-merging

### ISSUE THREAD

**Issue by sooyonee** [tanganke/fusion_bench#43]
Hi, I am interested in your unseen dataset experiment and trying to retrieve Table 4. I could retrieve accuracy for task arithmetic, however, I'm having difficulty retrieving the result for TIES-merging. Can you share your bash file and dataset setup for experiment in Table 4? Thanks!

**Comment by tanganke** [comment]
We do not have a model pool class to handle open_clip models yet because it looks like that a specific version of `open_clip` package is required to load models from piclkled binary checkpoint. 
But it is possible to merge open_clip vision models using FusionBench.

1. Import algorithm class in your own code:

```python
# load open_clip vision models
pretrained_model = load_open_clip_model(...) # this loads an nn.Module object
# load task-specific models
task_a_model = load_open_clip_model(...)
task_b_model = load_open_clip_model(...)
...
task_n_model = load_open_clip_model(...)
```

You can import an algorithm class from FusionBench and merge:

```python
from fusion_bench.method import TiesMergingAlgorithm

merged_model = TiesMergingAlgorithm(scaling_factor=... , threshold=..., ...).run({
  '_pretrained_': pretrained_model, 
  'task_a': task_a_model, 
  'task_b': task_b_model, 
  'task_n': task_n_model
})
```

2. Implement model pool and task pool class for openclip models.

You can refer to the implementation of `CLIPVisionModelPool` and `CLIPVisionModelTaskPool`.
These are major methods that need to be implemented:

```python
class YourModelPool:
    def load_model(self, ...):
        ...
    def load_processor(self, ...):
        ...
    def load_test_dataset(self, ...):
        # for test-time adaptation
        ...

class YourTaskPool:
    def evaluate(self, merged_model):
        ...

**Comment by tanganke** [comment]
Closing this. If you have any other questions, feel free to reopen it or create a new issue.
